Userland/Services/LookupServer/DNSPacket.cpp

```diff
@@ -144,17 +145,28 @@ Optional<DNSPacket> DNSPacket::from_raw_packet(const u8* raw_data, size_t raw_si
 
         offset += sizeof(DNSRecordWithoutName);
 
-        if (record.type() == T_PTR) {
+        switch (record.type()) {
+        case DNSRecordType::PTR: {
             size_t dummy_offset = offset;
             data = DNSName::parse(raw_data, dummy_offset, raw_size).as_string();
-        } else if (record.type() == T_A) {
+            break;
+        }
+        case DNSRecordType::A:
+            // Fall through
+        case DNSRecordType::TXT:
+            // Fall through
+        case DNSRecordType::AAAA:
+            // Fall through
+        case DNSRecordType::SRV:
             data = { record.data(), record.data_length() };
-        } else {
+            break;
+        default:
             // FIXME: Parse some other record types perhaps?
-            dbgln("data=(unimplemented record type {})", record.type());
+            dbgln("data=(unimplemented record type {})", (u16)record.type());
         }
+
         dbgln_if(LOOKUPSERVER_DEBUG, "Answer   #{}: name=_{}_, type={}, ttl={}, length={}, data=_{}_", i, name, record.type(), record.ttl(), record.data_length(), data);
-        packet.m_answers.empend(name, record.type(), record.record_class(), record.ttl(), data);
+        packet.m_answers.empend(name, record.type(), record.record_class(), record.ttl(), data, false);
```

Comment:
Why `false`?

Reply:
I've updated that to propagate the value from the DNS packet we received.